### DISCUSSION TITLE
Admin rest api response models

### DISCUSSION THREAD

**Discussion by qw-in** [SatelCreative/spylib#83] [Ideas]
I would love to have fully typed models for at least the most popular rest api endpoints!

For example products & orders would be super helpful!

```python
product = await token.product("123456789")

# Where product is a pydantic model

class Product(BaseModel):
    id: str
    title: str
    # ...
```

Notes:
- Type checker could catch issues - especially nice when jumping api versions (say `2022-01` to `2022-06`)
- Autocomplete & inline documentation :heart:

**Comment by hillairet** [comment]
I heard someone had found the Shopify APISpec ... there is this [datamodel-code-generator](https://github.com/koxudaxi/datamodel-code-generator/) that looks exciting ... I say it's not that far!

One think to plan for though is that currently the user can change the API version used for the Rest calls so having hardcoded types would be a form of dependency. I'm thinking the SPyLib version should reflect that and we may need to generate multiple versions somehow.
That's the challenge of this exciting idea.
